src/main/java/net/unicon/lti/service/lti/impl/DeepLinkServiceImpl.java

```diff
@@ -0,0 +1,170 @@
+
+    //Asking for a token with the right scope.
+    @Override
+    public List<DeepLinkDTO> getDeepLinks() {
+       List<ToolLink> toolLinkList = toolLinks.findAll();
+       List<DeepLinkDTO> deepLinkDTOS = new ArrayList<DeepLinkDTO>();
+       for (ToolLink toolLink:toolLinkList){
+           DeepLinkDTO deepLinkDTO = new DeepLinkDTO(toolLink.getToolLinkId(),toolLink.getTitle(), toolLink.getDescription());
+           deepLinkDTOS.add(deepLinkDTO);
+       }
+       return deepLinkDTOS;
+    }
+
+    private static String listMapToJson(List<Map<String, Object>> listMap) {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(listMap);
+        } catch (JsonProcessingException e) {
+            log.error(e.getMessage());
+            return "";
+        }
+    }
+    @Override
+    public DeepLinkJWTDTO generateDeepLinkJWT(List<String> deepLinkRequestIds, Jws<Claims> id_token) throws GeneralSecurityException, IOException {
+
+        //, String nonce, String deepLinkData, String deploymentId, String clientId, String iss
+
+        Date date = new Date();
+        Key toolPrivateKey = OAuthUtils.loadPrivateKey(ltiDataService.getOwnPrivateKey());
+        DeepLinkJWTDTO deepLinkJWTDTO = new DeepLinkJWTDTO();
+
+        List<Map<String,Object>> jsonDeepLinks = getDeepLinksJSON(deepLinkRequestIds);
+        deepLinkJWTDTO.setJSONString(jsonDeepLinks);
+
+        String jwt = Jwts.builder()
+                .setHeaderParam(LtiStrings.TYP, LtiStrings.JWT)
+                .setHeaderParam(LtiStrings.KID, TextConstants.DEFAULT_KID)
+                .setHeaderParam(LtiStrings.ALG, LtiStrings.RS256)
+                .setIssuer(id_token.getBody().getAudience())//Client ID
+                .setAudience(id_token.getBody().getIssuer())
+                .setExpiration(DateUtils.addSeconds(date, 3600)) //a java.util.Date
+                .setIssuedAt(date) // for example, now
+                .claim(LtiStrings.LTI_NONCE, id_token.getBody().get(LtiStrings.LTI_NONCE, String.class))
+                .claim(LtiStrings.LTI_AZP, id_token.getBody().getIssuer())
+                .claim(LtiStrings.LTI_DEPLOYMENT_ID, id_token.getBody().get(LtiStrings.LTI_DEPLOYMENT_ID, String.class))
+                .claim(LtiStrings.LTI_MESSAGE_TYPE, LtiStrings.LTI_MESSAGE_TYPE_DEEP_LINKING_RESPONSE)
+                .claim(LtiStrings.LTI_VERSION, LtiStrings.LTI_VERSION_3)
+                .claim(LtiStrings.LTI_DATA, id_token.getBody().get(LtiStrings.DEEP_LINKING_SETTINGS, Map.class).get(LtiStrings.DEEP_LINK_DATA))
+                .claim(LtiStrings.LTI_CONTENT_ITEMS, jsonDeepLinks)
+                .signWith(SignatureAlgorithm.RS256, toolPrivateKey)  //We sign it
+                .compact();
+
+        deepLinkJWTDTO.setJWTString(jwt);
+        return deepLinkJWTDTO;
+    }
+
+    private List<Map<String,Object>> getDeepLinksJSON(List<String> deepLinkRequestIds) {
```

Comment:
Could we change the name of this method to `createDeepLinksJson` so that it matches the format of the one below but still shows that this is the plural version?